File: php/features.php
<?php
declare(strict_types=1);

// QuanhauApiDocs SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class QuanhauApiDocsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new QuanhauApiDocsBaseFeature();
            case "test":
                return new QuanhauApiDocsTestFeature();
            default:
                return new QuanhauApiDocsBaseFeature();
        }
    }
}

File: py/quanhauapidocs_sdk/features.py
# QuanhauApiDocs SDK feature factory

from quanhauapidocs_sdk.feature.base_feature import QuanhauApiDocsBaseFeature
from quanhauapidocs_sdk.feature.test_feature import QuanhauApiDocsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: QuanhauApiDocsBaseFeature(),
        "test": lambda: QuanhauApiDocsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
